GG/src/ListBox.cpp

```diff
@@ -431,12 +426,18 @@ void ListBox::Row::SetColAlignments(const std::vector<Alignment>& aligns)
     if (aligns == m_col_alignments)
         return;
 
-    m_col_alignments = aligns;
+    m_col_alignments.resize(m_cells.size(), ALIGN_NONE);
+
+    for (std::size_t i = 0; i < m_col_alignments.size(); ++i) {
```

Comment:
why?

Reply:
For any elements missing from `aligns`, the value of `m_col_alignments` is either unchanged or initialized to `ALIGN_NONE`.
If the size of `aligns` exceeds `m_cells`, the excess elements are ignored.

Reply:
would be simpler if you kept `m_col_alignments = aligns;` and then shrank or expanded `m_col_alignments` to be the same size as `m_cells`. The extra uses of `.at(i)` and different varieties of bounds checking are confusing to read.

Reply:
Ok